### PR TITLE
Update to include new TreeSitter capture names

### PR DESCRIPTION
**Pull request by JulesNP**
This update adds the new capture group names from the latest TreeSitter update (https://github.com/nvim-treesitter/nvim-treesitter/commit/3f978f85b55330217bceec53dd54cf33f3541076). As a backwards compatibility measure, the old capture group names are not yet being removed.

### CODE REVIEW

File: lua/gruvbox.lua [hunk 1]
```diff
@@ -991,9 +1004,35 @@ local function get_groups()
...
+    ["@markup.link.uri"] = { link = "Underlined" },
+    ["@markup.math"] = { link = "Special" },
+    ["@markup.environment"] = { link = "Macro" },
+    ["@markup.environment.name"] = { link = "Type" },
+    ["@markup.link"] = { link = "Constant" },
+    ["@markup.link.label"] = { link = "SpecialChar" },
+    ["@markop.list"] = { link = "Delimiter" },
+    ["@comment.todo"] = { link = "Todo" },
+    ["@comment.todo.checked"] = { link = "GruvboxGreen" },
+    ["@comment.todo.unchecked"] = { link = "GruvboxGray" },
```
Comment (ObserverOfTime) [inline lua/gruvbox.lua:1028]:
```suggestion
    ["@markup.list"] = { link = "Delimiter" },
    ["@markup.list.checked"] = { link = "GruvboxGreen" },
    ["@markup.list.unchecked"] = { link = "GruvboxGray" },
    ["@comment.todo"] = { link = "Todo" },
```

File: lua/gruvbox.lua [hunk 2]
```diff
@@ -991,9 +1004,35 @@ local function get_groups()
...
+    ["@comment.note"] = { link = "SpecialComment" },
+    ["@comment.warning"] = { link = "WarningMsg" },
+    ["@comment.danger"] = { link = "ErrorMsg" },
```
Comment (ObserverOfTime) [inline lua/gruvbox.lua:1031]:
```suggestion
    ["@comment.error"] = { link = "ErrorMsg" },
```

File: lua/gruvbox.lua [hunk 3]
```diff
@@ -956,25 +956,38 @@ local function get_groups()
     ["@punctuation.special"] = { link = "Delimiter" },
     ["@string"] = { link = "String" },
     ["@string.regex"] = { link = "String" },
+    ["@string.regexp"] = { link = "String" },
     ["@string.escape"] = { link = "SpecialChar" },
     ["@string.special"] = { link = "SpecialChar" },
+    ["@string.special.symbol"] = { link = "Identifier" },
+    ["@string.special.uri"] = { link = "Underlined" },
```
Comment (ObserverOfTime) [inline lua/gruvbox.lua:963]: Missing `@string.special.path`.

Comment (JulesNP) [reply lua/gruvbox.lua:964]: Thanks, made your suggested changes. For `@string.special.path`, I've set it to be a link to "Underlined" also.

File: lua/gruvbox.lua [hunk 4]
```diff
@@ -956,25 +956,39 @@ local function get_groups()
     ["@punctuation.special"] = { link = "Delimiter" },
     ["@string"] = { link = "String" },
     ["@string.regex"] = { link = "String" },
+    ["@string.regexp"] = { link = "String" },
     ["@string.escape"] = { link = "SpecialChar" },
     ["@string.special"] = { link = "SpecialChar" },
+    ["@string.special.filename"] = { link = "Underlined" },
```
Comment (ObserverOfTime) [inline lua/gruvbox.lua:962]:
```suggestion
    ["@string.special.path"] = { link = "Underlined" },
```

File: lua/gruvbox.lua [hunk 5]
```diff
@@ -956,25 +956,39 @@ local function get_groups()
...
+    ["@string.special.symbol"] = { link = "Identifier" },
+    ["@string.special.uri"] = { link = "Underlined" },
```
Comment (ObserverOfTime) [inline lua/gruvbox.lua:964]:
```suggestion
    ["@string.special.url"] = { link = "Underlined" },
```

File: lua/gruvbox.lua [hunk 6]
```diff
@@ -991,9 +1005,35 @@ local function get_groups()
...
     ["@constant.builtin"] = { link = "Special" },
     ["@constant.macro"] = { link = "Define" },
+    ["@markup"] = { link = "GruvboxFg1" },
+    ["@markup.strong"] = { bold = config.bold },
+    ["@markup.emphasis"] = { italic = config.italic.emphasis },
+    ["@markup.underline"] = { underline = config.underline },
+    ["@markup.strike"] = { strikethrough = config.strikethrough },
+    ["@markup.heading"] = { link = "Title" },
+    ["@markup.raw"] = { link = "String" },
+    ["@markup.link.uri"] = { link = "Underlined" },
```
Comment (ObserverOfTime) [inline lua/gruvbox.lua:1020]:
```suggestion
```

File: lua/gruvbox.lua [hunk 7]
```diff
@@ -991,9 +1005,35 @@ local function get_groups()
...
+    ["@markup.math"] = { link = "Special" },
+    ["@markup.environment"] = { link = "Macro" },
+    ["@markup.environment.name"] = { link = "Type" },
+    ["@markup.link"] = { link = "Constant" },
```
Comment (ObserverOfTime) [inline lua/gruvbox.lua:1024]:
```suggestion
    ["@markup.link"] = { link = "Underlined" },
```

**Review by ObserverOfTime — APPROVED** [review]
LGTM but I'm not the maintainer.

**Comment by JulesNP** [comment]
@ellisonleao, are you OK with merging these changes?

**Comment by ellisonleao** [comment]
hey @JulesNP @ObserverOfTime thanks for the PR, i will look on them asap

**Pull request merged**